warehouse/slave.go

```diff
@@ -485,6 +485,31 @@ func processStagingFile(job Payload, workerIndex int) (loadFileUploadOutputs []l
 			columnType := columnInfo.Type
 			columnVal := columnInfo.Value
 
+			if job.DestinationType == warehouseutils.CLICKHOUSE && model.SchemaType(columnType) == model.BooleanDataType {
```

Comment:
Did we discover this during testing?

Reply:
We are mapping boolean to UInt8 datatype in clickhouse. In our previous load table strategy, when we try to insert the data, we used to convert true/false to 0/1 in [here](https://github.com/rudderlabs/rudder-server/blob/b7aa80003e237ddda370dff125898d4ac52ea820/warehouse/clickhouse/clickhouse.go#L504).

Reply:
Since now we are using COPY command which does the part of loading the data, We want to make sure that the load files generated are compatible with the table schema.

Reply:
We are adding this check here to  convert true/false to 0/1 directly in the load files itself.

Reply:
For the old load strategy, we are not handling both cases.
1. true/false
2. 0/1